src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/DefaultMonitorService.java

```diff
@@ -116,6 +116,16 @@ public void stopMonitoring(MonitorConnectionContext context) {
     this.threadContainer.getMonitor(node).stopMonitoring(context);
   }
 
+  @Override
+  public void failoverMonitor(Set<String> nodeKeys) {
```

Comment:
Any recommendation for a better method name?

Reply:
What does this method do? :)

Reply:
When failover occurs, we want the monitor to start monitoring the new node. 
We have to remove all the node keys pointing to the old node first. This method removes all the references to the old node and put the monitor in a "waiting queue" so we don't lose the monitor.

When `startMonitoring` is called later when a new node has been found, we will take the monitor from that waiting queue.